### ISSUE TITLE
Add a logger for Sacred

### ISSUE THREAD

**Issue by agrinh** [PyTorchLightning/lightning-bolts#443]
### Motivation
Sacred also needs some love as seen here: https://github.com/PyTorchLightning/pytorch-lightning/issues/438

### Pitch
Add a logger compatible with [Sacred](http://github.com/IDSIA/sacred).

### Alternatives
Not clear to me whether this should go into lightningbolts or lightning, I'd personally be happy with either but it was previously suggested that it might fit here.

### Additional context
Previously discussed here: https://github.com/PyTorchLightning/pytorch-lightning/issues/438

**Comment by akihironitta** [comment]
@agrinh Thank you for submitting an issue! As I looked through the linked issue, @expectopatronum first tried adding the logger, and the PR https://github.com/PyTorchLightning/pytorch-lightning/pull/656 was closed without getting merged. I think it'd be great to have the logger in Bolts. Would you be interested in submitting a PR?

**Comment by agrinh** [comment]
@akihironitta Hey, sorry for the delay!

I had some real trouble getting the referenced PR working in DDP since the sacred objects cannot be pickled. I'm not sure if there's a _nice_ way of solving this. I ended up writing an implementation that holds the sacred object in a manager thread in the main process, and accepts metrics to log over a queue from the other processes. It works, but seems error prone and annoying to maintain.

I'd be happy to contribute what I have if you still think it's useful, but I ended up moving over to better supported experiment loggers and don't have much more time to build something more stable. Let me know and I can open a PR.
